Update uniwersytet-gdanski-kulturoznawstwo.csl

kjklosowski opened the pull request:
Two changes here:
- Added `author` and `author-bibliography` in corresponding `type="software"` to reflect a situation where the software (or a video game) is authored by a programmer, or a group of programmers, instead of a sofware house/studio, as per internal request I received. In Zotero, a `"Programmer"` field is just `author` in CSL, so this is a fairly simple change, but will be reflected in rendering and sorting (these styles sort by authors and then titles if no author present).
- Slightly rephrased `summary` for clarity.

I'm the original author of the style (#7549, #7547).

Thanks!

POBrien333 commented on uniwersytet-gdanski-kulturoznawstwo-autor-rok.csl:
```diff
@@ -15,8 +15,8 @@
+    <updated>2025-09-24T15:17:52+00:00</updated>
     <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
   </info>
   <locale xml:lang="pl">
```
You can delete this whole "locale" section since the et al. term is already like this in our pl-PL locale: https://github.com/citation-style-language/locales/blob/master/locales-pl-PL.xml

POBrien333 reviewed (changes requested):
This style can be a lot more simplified:
1. you can just call author, title, [conditional for all item types]. 
That way any item types receives an author+title no matter what. Further bibliographic details can then be output via your conditional.
2. You have a lot of conditionals that test for a variable to then output said variable. These are all obsolete and you can remove those.

kjklosowski commented:
Hi @POBrien333! Thanks for looking at the code.
Yeah, the redundant if-statements are definietly overengineered, I regret ever doing it that way and we actually had a conversation about it during initial PR.

I have optimized the code greatly, removed all redundant if-statements, improved group-delimiters logic in some places. The only ifs that remained are there to guard against orphan terms showing up, when the user (for whatever reason) decides to leave a particular variable empty. I would prefer to avoid that, as this style can be used for many different applications. So the terms remaing guarded.

As for locale, I've received yet another request to change these terms to their Polish counterparts. The documentation permits both (provided that one is used consistently!), but in the given Department the Polish ones are used much more frequently, hence the change.

I have not ventured into your first point `"you can just call author, title, [conditional for all item types]."`. I mean, it can be done, it would decrease the size of file, but it would not decrease the amount of times each object is called (in a given citation object). I know what you mean by that, I can play around with it if you want, but **for me** it would make the file less readable, and I'm personally not that into it.

The ifs were most important, and they are gone!

kjklosowski commented on uniwersytet-gdanski-kulturoznawstwo-przypis.csl:
```diff
@@ -36,36 +36,34 @@
     <names variable="author">
       <name delimiter-precedes-last="always" initialize="false" initialize-with=". " name-as-sort-order="all" sort-separator=" "/>
-      <et-al font-style="italic"/>
+      <et-al font-style="normal"/>
```
I think `font-style="normal"` can be removed also?

POBrien333 replied:
That whole line can be removed since nothing is "done" to the et-al.

kjklosowski replied:
Oh I thought it would impact et-al behavior. I will remove all of them

POBrien333 commented:
To remove the validation error:
Can you go here and click on "Sync fork"?
https://github.com/kjklosowski/styles/tree/patch-1

POBrien333 commented on uniwersytet-gdanski-kulturoznawstwo-przypis.csl:
```diff
@@ -15,15 +15,15 @@
-      <term name="op-cit">op. cit.</term>
+      <term name="ibid">tamże</term>
+      <term name="et-al">i in.</term>
+      <term name="op-cit">dz. cyt.</term>
```
Our polish locale probably just still has the English term in it. Would you say that "dz. cit." should be the general translation?
https://github.com/citation-style-language/locales/blob/208128470ef09fba9fdb5bae5fb442a91e364668/locales-pl-PL.xml#L45

kjklosowski replied:
Actually the polish locale used to include the "i in." but we changed it to latin with Brenton, as that is generally more common in Poland, but in that style Polish counterparts could work as well. See: [Update locales-pl-PL.xml](https://github.com/citation-style-language/locales/pull/353).

I can just remove the local section and stick with default global locale if that's what you prefer, it's not that important.

POBrien333 commented on uniwersytet-gdanski-kulturoznawstwo-autor-rok.csl:
```diff
@@ -16,12 +16,16 @@
   <locale xml:lang="pl">
     <terms>
       <term name="et-al">i in.</term>
+      <term name="number" form="short"> <!-- The default-locale term is currently broken -->
```
Is this because you're using the Code/VisualEditor at https://editor.citationstyles.org/codeEditor/? 

You can ignore that and remove this. If you test it out inside the Zotero editor you'll see it works. The CSL Code/VisualEditor needs an update, that's why.

kjklosowski replied:
I've been just writing a comment about this.

Actually, Zotero behaves the same way, so I'm still concerned about it..

POBrien333 replied:
You need to call `<label variable="number" form="short"/>`

kjklosowski replied:
Still the same behavior inside Zotero with label variable :/

kjklosowski commented:
> It looks like you have changed the settings that we as admins can't change the code in your PR without making PRs on your branch. Can you change that?

Changed it. Thanks @POBrien333 for your assistance. The issue is now resolved!

I've removed these terms :)

POBrien333 commented on uniwersytet-gdanski-kulturoznawstwo-przypis.csl:
```diff
@@ -242,167 +185,101 @@
+                  <group delimiter=", ">
+                    <text macro="author"/>
+                    <text macro="title"/>
+                    <text variable="container-title" font-style="italic" prefix="[hasło w:] "/>
```
what does the "haslo w" mean? 
We try to use terms/labels where possible and not hardcode text like this.

POBrien333 commented on uniwersytet-gdanski-kulturoznawstwo-przypis.csl:
```diff
@@ -242,167 +185,101 @@
-                      </if>
-                    </choose>
+                    <text macro="locator"/>
                     <text macro="title" prefix="[hasło: " suffix="]"/>
```
same as above

kjklosowski replied:
A leftover I think

POBrien333 commented on uniwersytet-gdanski-kulturoznawstwo-przypis.csl:
```diff
@@ -555,117 +373,71 @@
-                  <name delimiter-precedes-last="always" initialize="false" initialize-with=". " name-as-sort-order="all" sort-separator=" "/>
-                  <et-al font-style="italic"/>
-                </names>
+              <if match="any" variable="collection-title">
```
Can you explain what you are trying to do with this test? It seems a bit odd to me.

kjklosowski replied:
Actually all ifs you have deleted in your commit preceded terms and labels. It's the only change I'm against as it can cause the label being present when there is no actual number following it.

I actually had an issue where the book was a part of a collection that had a name but had no number, the 'nr' label was printed anyways, making this unusable.

I think that all labels should have these ifs before for that particular reason. Can you reconsider that and let me readd these? There would be only a couple of them.

POBrien333 replied:
As you see, I've put them in a group. A group is like a conditional and it will only print out the label if the variable is also present. Test it out.

POBrien333 reviewed (changes requested):
Hi
I made some improvements. See how I created the "collection" macro that I call everywhere.
I also removed some more superfluous conditionals.
There is some remaining comments.
If you can, please then make these changes also in the other file. :)

kjklosowski commented:
Hi @POBrien333! Thanks for your contribution to my style.
I've answered you in these discussions you opened :)

POBrien333 commented:
I've merged the style now and it'll show up on zotero.org/styles within an hour or so. You can then use that link to share the style out.
If you spot any issues, of course you can make a new PR.
